app/api/resources/organizations.py

```diff
@@ -211,139 +257,158 @@ class CreateProgram(Resource):
         end date, description, target skills, target candidate selected from any of
-        the gender (value as per Gender enum), age (Age enum), ethnicity (Ethnicity enum), 
-        sexual orientation (SexualOrientation enum), religion (Religion enum), physical 
+        the gender (value as per Gender enum), age (Age enum), ethnicity (Ethnicity enum),
+        sexual orientation (SexualOrientation enum), religion (Religion enum), physical
         ability (PhysicalAbility enum), mental ability (MentalAbility enum), socio economic
         (SocioEEconomic enum), highest education (HighestEducation enum), years of experience
-        (YearsOfExperience enum), or other which value is self-defined by organization, payment 
-        currency code, payment amount, contact type (value as per ContactType enum), zone 
-        (with value as per Zone Enum value), mentee's responsibility, mentor's responsibility, 
-        organization's responsibility, mentee's requirements, mentor's requirements, 
-        resources provided, contact person's name, contact person's department, program address, 
-        contact person's phone, contact person's mobile, contact person's email, program webiste, 
+        (YearsOfExperience enum), or other which value is self-defined by organization, payment
+        currency code, payment amount, contact type (value as per ContactType enum), zone
+        (with value as per Zone Enum value), mentee's responsibility, mentor's responsibility,
+        organization's responsibility, mentee's requirements, mentor's requirements,
+        resources provided, contact person's name, contact person's department, program address,
+        contact person's phone, contact person's mobile, contact person's email, program webiste,
         irc channel, tags, status (with value as per ProgramStatus Enum value), and creation date.
         """
         token = request.headers.environ["HTTP_AUTHORIZATION"]
 
         is_wrong_token = validate_token(token)
-
+        status_code = HTTPStatus.BAD_REQUEST
         if not is_wrong_token:
             data = request.json
             if not data:
-                return messages.NO_DATA_FOR_UPDATING_PROFILE_WAS_SENT, HTTPStatus.BAD_REQUEST
+                return messages.NO_DATA_FOR_UPDATING_PROFILE_WAS_SENT, status_code
 
-            is_field_valid = expected_fields_validator(data, update_program_request_model)
+            is_field_valid = expected_fields_validator(
+                data, update_program_request_model
+            )
             if not is_field_valid.get("is_field_valid"):
-                return is_field_valid.get("message"), HTTPStatus.BAD_REQUEST
-            
+                return is_field_valid.get("message"), status_code
+
             is_not_valid = validate_update_program(data)
             if is_not_valid:
-                return is_not_valid, HTTPStatus.BAD_REQUEST
+                if is_not_valid == PROGRAM_NAME_ALREADY_USED:
```

Comment:
This is going to introduce a race condition. Instead you should try to update the program and catch the integrity error that the database throws.

Reply:
I agree that generally, this can be prone to a race condition, but it depends on the transactional behavior of the ORM and the isolation level.

Reply:
@NenadPantelic , does this mean you will make the changes as suggested by @epicadk?

Reply:
Yes, I will. Exception handling is safer, cause it prevents the possible race condition (between the check and this insert, another process can insert the record).